hs-bindgen/src-internal/HsBindgen/C/Predicate.hs

```diff
@@ -36,11 +37,8 @@ data Predicate =
     -- Used to define the 'Semigroup' instance.
   | SelectIfBoth Predicate Predicate
 
-    -- | Check if the definition is from the \"main\" file
-    --
-    -- Corresponds directly to @clang_Location_isFromMainFile@ in @libclang@;
-    -- <https://clang.llvm.org/doxygen/group__CINDEX__LOCATIONS.html#gacb4ca7b858d66f0205797ae84cc4e8f2>.
-  | SelectFromMainFile
+    -- | Include definitions in main files (and not in included files)
+  | SelectFromMainFiles
```

Comment:
I decided against adding `Set SourceFile` here. The reason is just that the option parsind would be more difficult, because we already need to know the main files then (or we create an intermediate option datatype). @edsko